### PR TITLE
fix(DB/SmartAI): Demolitionist Legoso weird behavior

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
## Issues Addressed:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/4092

## Tests Performed:
- Tested in game

## How to Test the Changes:

- `.go c id 17986`
- aggro an Enraged Ravager
- bring it to the Demolitionist
- verify that he does not cast lightning shields

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1609803806703890000.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1609803806703890000');
+
+-- add initial delay of half a second to IC Update - Cast Spell Lightning Shield
+DELETE FROM `smart_scripts` WHERE (`entryorguid` = 17982) AND (`source_type` = 0) AND (`id` IN (4));
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(17982, 0, 4, 0, 0, 0, 100, 0, 500, 500, 15000, 15000, 0, 11, 325, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'IC Update - Cast Spell Lightning Shield');
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1609803806703890000.sql:6]: You dont use comments from Keira? Missing NPC name

**Comment by Hacaw** [comment]
@FrancescoBorzi just tested it. 
It works, the npc no longer casts randomly lightning shields.

He only has a little stutter, seems like he wants to replay some animation, but it's not noticibile. I tried to compare it with Vindicator Corin (the npc near him) but she is no smart-ai.

**Review by Hacaw — CHANGES_REQUESTED** [review]
@FrancescoBorzi just tested it. 
It works, the npc no longer casts randomly lightning shields when you bring aggroed monsters near him.

But after testing it more I found more issues:

1) He only has a little stutter after you bring a ravager to him, seems like he wants to replay some animation, but it's not noticeable. I tried to compare it with Vindicator Corin (the npc near him) but she is no smart-ai. 

2) When playing .q a 9761 -> q c 9761 and take the quest "Ending their World" (escort quest) he doesn't talk. Check here https://www.youtube.com/watch?v=lARzK9CLu24 when the player accepts the quest.

3) He casts healing wave when he is almost full hp ( < 100%) when in combat with belfs.

4) Also while in the escort-quest he runs without waiting you until the distance is so big he disappears ( I think I tested a similar issue with a quest in Eastern Plaguelands will have to recheck)

5) He is not attacking the ravagers, only the belfs.

6) Also while in the escort-quest if you have time to aggro a raveger, he will keep coming back and forth and cast lightning shields (but it doesn't look like the orbs are applied to him)

**Comment by FrancescoBorzi** [comment]
@Hacaw thanks for your feedback. These additional issues are not introduced by this PR and thus should be reported in a new issue.

This can be merged as it solved the original reported issue without adding new issues.

File: data/sql/updates/pending_db_world/rev_1609803806703890000.sql (hunk 1 above)
Comment (wizzymore) [reply data/sql/updates/pending_db_world/rev_1609803806703890000.sql]: :O

**Pull request merged**